influxql/functions.go

```diff
@@ -357,6 +357,70 @@ func (r *IntegerMovingAverageReducer) Emit() []FloatPoint {
 	}
 }
 
+// FloatCumulativeSumReducer cumulates the values from each point.
+type FloatCumulativeSumReducer struct {
+	curr    FloatPoint
+	emitted bool
+}
+
+// NewFloatCumulativeSumReducer creates a new FloatCumulativeSumReducer.
+func NewFloatCumulativeSumReducer() *FloatCumulativeSumReducer {
+	return &FloatCumulativeSumReducer{
+		curr: FloatPoint{Nil: true},
+	}
+}
+
+func (r *FloatCumulativeSumReducer) AggregateFloat(p *FloatPoint) {
+	if !r.curr.Nil && r.curr.Time == p.Time {
```

Comment:
I think we shouldn't skip points in this case. The reason that derivative does it is it needs to use the elapsed time and it would create a divide by 0 otherwise. Since this is just a sum I would rather it simply sum the values and nothing else, similar to moving_avg.